test/unit/gateways/secure_pay_au_test.rb

```diff
@@ -27,6 +28,18 @@ def test_supported_card_types
     assert_equal %i[visa master american_express diners_club jcb], SecurePayAuGateway.supported_cardtypes
   end
 
+  def test_endpoint
+    xml_request = @gateway.send(:build_request, :purchase, Builder::XmlMarkup.new.target!)
+    doc = Nokogiri::XML(xml_request)
+    assert_equal 'xml-4.2', doc.xpath('SecurePayMessage/MessageInfo/apiVersion').text
+  end
+
+  def test_periodic_endpoint
+    xml_request = @gateway.send(:build_periodic_request, Builder::XmlMarkup.new.target!)
```

Comment:
You don't really have to use nokogiri to check for the api version. For example if you go to [test_periodic_payment_submits_order_id](https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/secure_pay_au_test.rb#L82) you can add `assert_match(/<apiVersion>spxml-3.0<\/apiVersion>/, data)` to check the Periodic API Version